fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -581,28 +598,198 @@ void basicGroupedPartitionedTest() {
+                scanQuery,
+                continuation,
+                ExecuteProperties.newBuilder().setSkip(amountToSkip).setReturnedRowLimit(1).build().asScanProperties(false));
+
+        List<IndexEntry> entries = cursor.asList().join();
+        assertEquals(1, entries.size());
+        RecordCursorResult<IndexEntry> lastResult = cursor.onNext().get();
+        assertEquals(RecordCursor.NoNextReason.RETURN_LIMIT_REACHED, lastResult.getNoNextReason());
+        LuceneContinuationProto.LuceneIndexContinuation parsed = LuceneContinuationProto.LuceneIndexContinuation.parseFrom(lastResult.getContinuation().toBytes());
+        // we stopped in the expected partition
+        assertEquals(expectedPartitionId, parsed.getPartitionId());
+        assertEquals(expectedDocId, entries.get(0).getPrimaryKey());
+        return lastResult.getContinuation().toBytes();
+    }
+
+    @Test
+    void repartitionSyntheticGroupedTest() {
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_PARTITION_HIGH_WATERMARK, 10)
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 6)
+                .build();
+
+        final int totalDocCount = 20;
+        Consumer<FDBRecordContext> schemaSetup = context -> openRecordStore(context, LuceneIndexTest::joinedPartitionedLuceneIndexMetadataHook);
+
+        // create/save documents
+        try (FDBRecordContext context = openContext(contextProps)) {
+            schemaSetup.accept(context);
+            long start = Instant.now().toEpochMilli();
+            for (int i = 0; i < totalDocCount; i++) {
+                TestRecordsTextProto.ComplexDocument cd = TestRecordsTextProto.ComplexDocument.newBuilder()
+                        .setGroup(42)
+                        .setDocId(1000L + i)
+                        .setIsSeen(true)
+                        .setTimestamp(start + i * 100)
+                        .build();
+                TestRecordsTextProto.SimpleDocument sd = TestRecordsTextProto.SimpleDocument.newBuilder()
+                        .setGroup(42)
+                        .setText("Four score and seven years ago our fathers brought forth")
+                        .build();
+                recordStore.saveRecord(cd);
+                recordStore.saveRecord(sd);
+            }
+            commit(context);
+        }
 
+        // initially, all documents are saved into one partition
+        List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = getPartitionMeta(JOINED_INDEX,
+                Tuple.from(42), contextProps, schemaSetup);
+        assertEquals(1, partitionInfos.size());
+        assertEquals(totalDocCount, partitionInfos.get(0).getCount());
+
+        // run re-partitioning
+        explicitMergeIndex(JOINED_INDEX, contextProps, schemaSetup);
+        partitionInfos = getPartitionMeta(JOINED_INDEX,
```

Comment:
It seems like this should also assert that the right documents end up in the right group